### PR TITLE
test(session): await on session cleanup to prevent flakiness (#3188)

### PR DESCRIPTION
**Pull request by Kroustille**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.
>
> ℹ️ **Feature environment** — A dedicated preview environment (`https://dev-pr-{number}.hub.staging.filigran.io`) is automatically deployed when CI passes. Add the `skip-feature-env` label to opt out.

Some API tests are failing because we don't wait for session cleanup so it can be done after the test assertion.
This PR update the session cleanup function to return a promise and the tests await on that before asserting.

Nothing to test, only technical changes

## Related issues

No ticket.

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code
- [x] I added the `skip-feature-env` label if a feature environment is not needed for this PR

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [ ] Unit tests
- [x] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/backend/src/utils/error/error.code.ts [hunk 1]
```diff
@@ -97,7 +97,6 @@ export enum UnknownErrorCode {
   RemoveUserFromOrgaError = 'REMOVE_USER_FROM_ORGA_ERROR',
   RemoveUserFromPendingOrgaError = 'REMOVE_USER_FROM_PENDING_ORGA_ERROR',
   AcceptUserInPendingOrgaError = 'ACCEPT_USER_IN_PENDING_ORGA_ERROR',
-  EditUserSessionError = 'EDIT_USER_SESSION_ERROR',
```
Comment (jpkha) [inline apps/backend/src/utils/error/error.code.ts:100]: Why do you remove this error?  Should not be related to the issue?

Comment (Kroustille) [reply apps/backend/src/utils/error/error.code.ts:100]: I removed this throwing error because this way, the other sessions are cleaned up except the ones in error. A log is still written to know something bad happened.

What do you think ? 🙂

File: apps/backend/src/session-store-manager.test.ts [hunk 2]
```diff
@@ -369,7 +402,7 @@ describe('sessionStoreManager - Configuration-based Store Selection', () => {
         last_name: 'Test',
         organizations: [],
         role: { id: 'role-id', name: 'USER' },
-      } as UserWithOrganizationsAndRole;
+      } as unknown as UserWithOrganizationsAndRole;
```
Comment (jpkha) [inline apps/backend/src/session-store-manager.test.ts:405]: What does property we are missing in this one to be force to cast as unknown ?

Comment (Kroustille) [reply apps/backend/src/session-store-manager.test.ts:405]: It's missing `roles_portal`, `selected_org_capabilities`, `salt`, `password`, `selected_organization_id`, `picture`, `disabled`, `last_login` which is quite a lot and that's why it was easier to force cast it to the right type.

I can have a base user with this properties populated with dummy values if it's better 😉

**Review by jpkha — APPROVED** [review]
